### ISSUE TITLE
OFFICIAL_EPOCH_LENGTH = 1250

### ISSUE THREAD

**Issue by onvungocminh** [facebookresearch/dinov2#135]
Hi all, In the script you defined OFFICIAL_EPOCH_LENGTH = 1250. How is that?
How can I define OFFICIAL_EPOCH_LENGTH in my custom dataset?

**Comment by qasfb** [comment]
So this value is purely arbitrary and corresponds to the number of iterations to do a pass on a dataset of the size of ImageNet-1k with a batch size of 1024. We have observed in our experiments that what matters is the number of iterations rather than the number of passes on a dataset, so this value is simply a multiplier for other "epochs" values in the config files (such as the total number of training epochs, the number of LR warmup epochs etc).

**Comment by onvungocminh** [comment]
Hi @qasfb, many thank for you reply. For a better training, Should I set the value of OFFICIAL_EPOCH_LENGTH = number_images/batch_size or I increase the number of epochs?

**Comment by qasfb** [comment]
It's difficult to say, I think it really depends on your data, maybe start with the short vit-l/16 training setup and see what you get ?

**Comment by onvungocminh** [comment]
if I set the OFFICIAL_EPOCH_LENGTH < number_images/batch_size, does it mean that the network can not see all the images in the dataset?

**Comment by qasfb** [comment]
No, the OFFICIAL_EPOCH_LENGTH is simply a multiplier that converts "epochs" into a number of iterations, it does not have any effect on the data loading; the data loader will iterate over the dataset independently of epochs.

**Comment by onvungocminh** [comment]
Thank @qasfb, it is clear now.
